waitress/task.py

```diff
@@ -251,9 +251,11 @@ def close_on_finish():
         # if the server is used as a proxy.
         ident = self.channel.server.adj.ident
         if not server_header:
-            response_headers.append(('Server', ident))
+            if ident:
+                response_headers.append(('Server', ident))
         else:
             response_headers.append(('Via', ident))
```

Comment:
If ident is falsey here, we should set it to 'waitress'. We should not append an empty `Via` header.